libs/platform/experience/carousel-navigation/carousel-navigation.component.ts

```diff
@@ -123,13 +127,16 @@ export class CarouselNavigationComponent
    */
   protected initializeResizeObserver(): void {
     if (this.resizeObserver) return;
-    const throttleTime = this.hostElement.clientWidth
-      ? this.intersectionThrottleTime
-      : 0;
     this.resizeObserver = new ResizeObserver(
-      throttle(() => {
-        this.buildNavigation();
-      }, throttleTime)
+      throttle((entries: ResizeObserverEntry[]) => {
+        window.requestAnimationFrame((): void | undefined => {
+          if (!Array.isArray(entries) || !entries.length) return;
+          setTimeout(
+            () => this.buildNavigation(),
+            this.intersectionThrottleTime
```

Comment:
We throttle to avoid too many repaints, however, with this approach we're often missing the latest resize. To ensure we're always taken the latest resized version, we're adding a timeout before rendering the latest time. 

I'm doing a small refactor though, to align the various throttle times, as it's overly complex now.

Reply:
Sounds not very stable

Reply:
I think it is, since the throttle and the timeout are the same. If 1ms before the last throttle time we trigger the repaint, we will still have the latest repaint.